### PR TITLE
ROX-19056: Extend Scanner v4 tests

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

This PR for improving the existing scanner V4 CI tests takes care of three things:

1. It adds more installation scenarios to the scanner v4 tests.
2. To not reinvent the wheel when introducing new test cases: port the tests to bats.
3. To make it pass CI: the PR removes the standard PostClusterTest machinery because it doesn't seem to be a good fit this kind of installation scenario test. The PostClusterTest includes things like a teardown, which is already done as part of the individual installation test scenarios.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~~Unit test and regression tests added~~
- [x] ~~Evaluated and added CHANGELOG entry if required~~
- [x] ~~Determined and documented upgrade steps~~
- [x] ~~Documented user facing changes~~

If any of these don't apply, please comment below.

## Testing Performed

Just CI greenness.

### CODE REVIEW

File: scripts/test_helpers.bats [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env bats
 
-# Allow to run the tests locally provided that bats-helpers are installed in $HOME/bats-core
-bats_helpers_root="${HOME}/bats-core"
+# Allow to run the tests locally provided that bats-helpers are installed.
+# If $BATS_HELPERS_ROOT is set, an attempt is made to load them from that directory,
+# otherwise they are expected in the default location $HOME/bats-core.
+# This makes sure that current behaviour is unaltered and existing workflows are not broken.
+bats_helpers_root=${BATS_CORE_ROOT:-${HOME}/bats-core}
```
Comment (mclasmeier) [inline scripts/test_helpers.bats:7]: To ease local testing. Doesn't change existing behaviour.

File: tests/e2e/lib.sh [hunk 2]
```diff
@@ -618,16 +618,37 @@ collect_and_check_stackrox_logs() {
 # system tests against the same cluster.
 remove_existing_stackrox_resources() {
     info "Will remove any existing stackrox resources"
-
+    local psps_supported=false
```
Comment (mclasmeier) [inline tests/e2e/lib.sh:621]: The changes in this file are just reducing some "resource X not found" noise from the logs.

File: tests/e2e/run-scanner-v4.bats [hunk 3]
```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bats
```
Comment (mclasmeier) [inline tests/e2e/run-scanner-v4.bats:1]: The existing tests have been converted to bats.

File: tests/e2e/run-scanner-v4.bats [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+       run remove_existing_stackrox_resources
+    fi
+    test_case_no=$(( test_case_no + 1))
+}
+
+teardown() {
+    run remove_existing_stackrox_resources
+}
+
+@test "Upgrade from old Helm chart to HEAD Helm chart with Scanner v4 enabled" {
```
Comment (mclasmeier) [inline tests/e2e/run-scanner-v4.bats:67]: This is the previously existing test.

File: tests/e2e/lib.sh (hunk 2 above)
Comment (vladbologa) [reply tests/e2e/lib.sh:621]: Nit: I think that these changes are not needed for the Scanner V4 tests, so I'd extract them into their own PR.

File: tests/e2e/run-scanner-v4.bats [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        # shellcheck disable=SC2030,SC2031
+        export MAIN_IMAGE_TAG
+        # shellcheck disable=SC2030,SC2031
+        export ROX_SCANNER_V4_ENABLED=false
+        # shellcheck disable=SC2030,SC2031
+        export OUTPUT_FORMAT=helm
+        deploy_stackrox >&3
+    )
+    verify_scannerV2_deployed "stackrox"
+    verify_no_scannerV4_deployed "stackrox"
```
Comment (vladbologa) [inline tests/e2e/run-scanner-v4.bats:133]: nit
```suggestion
    verify_scannerV4_not_deployed "stackrox"
```

**Review by vladbologa — APPROVED** [review]

File: scripts/test_helpers.bats [hunk 6]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env bats
 
-# Allow to run the tests locally provided that bats-helpers are installed in $HOME/bats-core
-bats_helpers_root="${HOME}/bats-core"
+# Allow to run the tests locally provided that bats-helpers are installed.
+# If $BATS_HELPERS_ROOT is set, an attempt is made to load them from that directory,
```
Comment (vladbologa) [inline scripts/test_helpers.bats:4]:
```suggestion
# If $BATS_CORE_ROOT is set, an attempt is made to load them from that directory,
```

Comment (mclasmeier) [reply scripts/test_helpers.bats:4]: I figured I would do it this way to reduce scope creep, i.e. keep rest unchanged.

Comment (vladbologa) [reply scripts/test_helpers.bats:4]: I thought that's a typo in the comment, because below you check if `BATS_CORE_ROOT` exists

Comment (mclasmeier) [reply scripts/test_helpers.bats:4]: d'oh, thanks.

Comment (mclasmeier) [reply scripts/test_helpers.bats:4]: fixed

File: tests/e2e/lib.sh (hunk 2 above)
Comment (mclasmeier) [reply tests/e2e/lib.sh:621]: Done, will be done separate, thanks.

File: tests/e2e/run-scanner-v4.bats (hunk 5 above)
Comment (mclasmeier) [reply tests/e2e/run-scanner-v4.bats:133]: hope it's fine to keep the naming at least for now, I don't have strong feelings about it.

**Pull request merged**